kratos/utilities/normal_calculation_utils.h

```diff
@@ -167,7 +155,26 @@ class NormalCalculationUtils
       */
     void CalculateOnSimplex(ModelPart& rModelPart,
                             int Dimension)
-    {
+    {   
+        // Resetting the normals
+        const array_1d<double,3> zero = ZeroVector(3);
```

Comment:
i would have a specialized path for the serial case. in that case you don't need to use the IS_VISITED

i mean

~~~cpp
if(IsDistributed){
...version with is visited...}
else
{
version we had before
}
~~~

Reply:
Ok